clients/wallet/src/stellar_wallet.rs

```diff
@@ -374,12 +377,108 @@ impl StellarWallet {
 			operations,
 		)?;
 
-		let _ = self.cache.save_tx_envelope(envelope.clone())?;
-
 		self.submit_transaction(envelope).await
 	}
 }
 
+// Error handling methods for StellarWallet.
+impl StellarWallet {
+	pub async fn handle_error(&self, error: Error) -> Result<TransactionResponse, Error> {
```

Comment:
Maybe we could change the return type to bool here (or simply don't return anything?)

Reply:
You mean because we don't actually use the `TransactionResponse` that is returned sometimes?
I think it's still nice to return an error here. Maybe we could change it to `Result<(), Error>` if we are sure that the `TransactionResponse` will never be needed.